sdk/src/wallet/operations/reissue.rs

```diff
@@ -16,48 +16,69 @@ use crate::{
     wallet::{types::InclusionState, Error, Wallet},
 };
 
-const DEFAULT_REISSUE_UNTIL_INCLUDED_INTERVAL: u64 = 1;
-const DEFAULT_REISSUE_UNTIL_INCLUDED_MAX_AMOUNT: u64 = 40;
+// Time in milliseconds
+const DEFAULT_AWAIT_TX_ACCEPTANCE_INTERVAL: u64 = 500;
+const DEFAULT_AWAIT_TX_ACCEPTANCE_MAX_AMOUNT: u64 = 80;
 
 impl<S: 'static + SecretManage> Wallet<S>
 where
     Error: From<S::Error>,
     crate::client::Error: From<S::Error>,
 {
-    /// Reissues a transaction sent from the account for a provided transaction id until it's
-    /// included (referenced by a milestone). Returns the included block id.
-    pub async fn reissue_transaction_until_included(
+    /// Checks the transaction state for a provided transaction id until it's accepted. Interval in milliseconds.
+    /// Returns the block id that contains this transaction.
+    pub async fn await_transaction_acceptance(
         &self,
         transaction_id: &TransactionId,
         interval: Option<u64>,
         max_attempts: Option<u64>,
     ) -> crate::wallet::Result<BlockId> {
-        log::debug!("[reissue_transaction_until_included]");
+        log::debug!("[await_transaction_acceptance]");
 
-        let protocol_parameters = self.client().get_protocol_parameters().await?;
-        let transaction = self.data().await.transactions.get(transaction_id).cloned();
+        let transaction = self
+            .data()
+            .await
+            .transactions
+            .get(transaction_id)
+            .cloned()
+            .ok_or_else(|| Error::TransactionNotFound(*transaction_id))?;
 
-        if let Some(transaction) = transaction {
-            if transaction.inclusion_state == InclusionState::Confirmed {
-                return transaction.block_id.ok_or(Error::MissingParameter("block id"));
-            }
+        if transaction.inclusion_state == InclusionState::Accepted
+            || transaction.inclusion_state == InclusionState::Confirmed
+            || transaction.inclusion_state == InclusionState::Finalized
+        {
```

Comment:
would it make sense to add an `is_accepted` method (and maybe others), that returns true if `self == Accepted | Confirmed | Finalized` (pseudocode) since certain states imply others?

Reply:
Currently I think it's not worth it, so far this is the only place where it's like that and in other places you probably want to differentiate more

Reply:
okay